plutus-tx-plugin/src/PlutusTx/Plugin.hs

```diff
@@ -550,19 +553,27 @@ runCompiler moduleName opts expr = do
     -- GHC.Core -> Pir translation.
     pirT <- original <$> (PIR.runDefT annMayInline $ compileExprWithDefs expr)
     let pirP = PIR.Program noProvenance plcVersion pirT
-    when (opts ^. posDumpPir) . liftIO $
+    when (opts ^. posDumpPir) . liftIO $ do
         dumpFlat (void pirP) "initial PIR program" (moduleName ++ "_initial.pir-flat")
 
+    (dumpCert, closeCertHandle) <- liftIO getDumpCert
+    liftIO $ dumpCert (Text.pack (show . UPLC.AsCoq $ void pirT))
+
     -- Pir -> (Simplified) Pir pass. We can then dump/store a more legible PIR program.
-    spirP <- flip runReaderT pirCtx $ PIR.compileToReadable pirP
-    when (opts ^. posDumpPir) . liftIO $
+    spirP <- flip runReaderT pirCtx $ PIR.compileToReadable (liftIO . dumpCert) pirP
+    when (opts ^. posDumpPir) . liftIO $ do
         dumpFlat (void spirP) "simplified PIR program" (moduleName ++ "_simplified.pir-flat")
 
+
     -- (Simplified) Pir -> Plc translation.
-    plcP <- flip runReaderT pirCtx $ PIR.compileReadableToPlc spirP
+    plcP <- flip runReaderT pirCtx $ PIR.compileReadableToPlc (liftIO . dumpCert) spirP
     when (opts ^. posDumpPlc) . liftIO $
         dumpFlat (void plcP) "typed PLC program" (moduleName ++ ".tplc-flat")
 
+    -- For now, we only dump the PIR -> PLC ASTs for the certifier,
+    -- so the handle can be closed.
+    liftIO closeCertHandle
```

Comment:
Not extremely important, but it is better to use the `withHandle`-idiom here to make sure that the handle is closed in case of an exception thrown